extensions/amp-form/0.1/amp-form.js

```diff
@@ -525,18 +568,15 @@ export class AmpForm {
   }
 
   /**
-   * @param {!IArrayLike<!HTMLInputElement>} varSubsFields
    * @param {ActionTrust} trust
    * @private
    */
-  handleXhrSubmit_(varSubsFields, trust) {
-    this.setState_(FormState.SUBMITTING);
-    const varSubPromise = this.doVarSubs_(varSubsFields);
+  handleXhrSubmit_(trust) {
     let p;
     if (this.ssrTemplateHelper_.isSupported()) {
-      p = varSubPromise.then(() => this.handleSsrTemplate_(trust));
+      p = Promise.resolve().then(() => this.handleSsrTemplate_(trust));
```

Comment:
nit:
```js
// Do this...
const p = functionThatReturnsAPromise();

// ...instead of this
const p = Promise.resolve().then(() => functionThatReturnsAPromise())

```
You can save bytes and make the code cleaner by using the `Promise` from the function and omitting the `Promise.resolve()`.

Reply:
Oops, yeah totally, thanks! 😄